modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/IdentityProviderMgtServiceTestCase.java

```diff
@@ -360,7 +365,202 @@ public void testGetAllIdPs() throws Exception {
+    public void testGetPaginatedIdPs() throws Exception {
+
+        addIdpForPagination();
+        int pageNumber = 1;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+        Assert.assertNotNull(idpList);
+        log.info("All idp paginated list size : " + idpList.size());
+        if (idpList.size() > 0) {
+            IdentityProvider filteredIdp = null;
+            for (IdentityProvider idp : idpList) {
+                if (testIdpName.equals(idp.getIdentityProviderName()) || testIdpNameSearch
+                        .equals(idp.getIdentityProviderName())) {
+                    filteredIdp = idp;
+                }
+            }
+            Assert.assertNotNull(filteredIdp, "Added new idps not found in the idp list");
+        } else {
+            Assert.fail("Unable to find added identity provider");
+        }
+    }
+
+    @Test(priority = 5, groups = "wso2.is", description = "test getAllPaginatedIdPsInfo operation with invalid page number")
+    public void testGetPaginatedIdPsWithInvalidPageNumber() {
+
+        try {
+            int pageNumber = -1;
+            List<IdentityProvider> idpList = idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+            Assert.assertNull(idpList,
+                    "List of Idps found  while calling getAllPaginatedIdPsInfo() with invalid page number");
+        } catch (Exception e) {
+            Assert.assertEquals(e.getMessage(), invalidPageNumberErrorMessage,
+                    "Error while testing testGetPaginatedIdPsWithInvalidPageNumber() with invalid page number");
+
+        }
+    }
+
+    @Test(priority = 6, groups = "wso2.is", description = "test getAllPaginatedIdPsInfo operation with wrong page number")
+    public void testGetPaginatedIdPsWithIncorrectPageNumber() throws Exception {
+
+        int pageNumber = 2;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+        if (idpList.size() > 0) {
+            Assert.fail("List of Idps found while calling getAllPaginatedIdPsInfo() with wrong page number.");
+        }
+    }
+
+    @Test(priority = 7, groups = "wso2.is", description = "test getPaginatedIdPsInfo operation")
+    public void testGetAllPaginatedIdPsWithFilter() throws Exception {
+
+        String filter = "name sw \"Search\" and name ew \"Test\"";
+        int pageNumber = 1;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+        Assert.assertNotNull(idpList);
+        if (idpList.size() > 0) {
+            IdentityProvider filteredIdp = null;
+            IdentityProvider notFilteredIdp = null;
+            for (IdentityProvider idp : idpList) {
+                if (testIdpNameSearch.equals(idp.getIdentityProviderName())) {
+                    filteredIdp = idp;
+                } else if (testIdpName.equals(idp.getIdentityProviderName())) {
```

Comment:
as in this test we are checking whether the idp list contains the expected filtered IDP, this else-if part is not required.